vertx-web/src/main/java/io/vertx/ext/web/handler/sse/impl/EventSourceImpl.java

```diff
@@ -0,0 +1,137 @@
+package io.vertx.ext.web.handler.sse.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxException;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.ext.web.handler.sse.EventSource;
+import io.vertx.ext.web.handler.sse.SSEHeaders;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class EventSourceImpl implements EventSource {
+
+	private HttpClient client;
+	private boolean connected;
+	private String lastId;
+	private Handler<String> messageHandler;
+	private final Map<String, Handler<String>> eventHandlers;
+	private SSEPacket currentPacket;
+	private final Vertx vertx;
+	private final HttpClientOptions options;
+	private Handler<Void> closeHandler;
+
+	public EventSourceImpl(Vertx vertx, HttpClientOptions options) {
+		options.setKeepAlive(true);
+		this.vertx = vertx;
+		this.options = options;
+		eventHandlers = new HashMap<>();
+	}
+
+	@Override
+	public EventSource connect(String path, Handler<AsyncResult<Void>> handler) {
+		return connect(path, null, handler);
+	}
+
+	@Override
+	public EventSource connect(String path, String lastEventId, Handler<AsyncResult<Void>> handler) {
+		if (connected) {
+			throw new VertxException("SSEConnection already connected");
+		}
+		if (client == null) {
+			client = vertx.createHttpClient(options);
+		}
+		HttpClientRequest request = client.get(path, result -> {
+		  if (result.failed()) {
+		    handler.handle(Future.failedFuture(result.cause()));
+		    return;
+      }
+      HttpClientResponse response = result.result();
+			if (response.statusCode() != 200) {
+				handler.handle(Future.failedFuture(new VertxException("Could not connect EventSource, the server answered with status " + response.statusCode())));
+			} else {
+				connected = true;
+				response.handler(this::handleMessage);
+				if (closeHandler != null) {
+					response.endHandler(closeHandler);
```

Comment:
The `EventSource` should reconnect when connection is lost.

Reply:
Fixed, but not sure about the implementation.
In case of a reconnection, the `connectHandler`  is called another time. i did this on purpose, but not sure if it's the right thing to do.